zio-http/src/main/scala/zhttp/http/Headers.scala

```diff
@@ -37,11 +37,18 @@ final case class Headers(toChunk: Chunk[Header]) extends HeaderExtension[Headers
   /**
    * Converts a Headers to [io.netty.handler.codec.http.HttpHeaders]
    */
-  private[zhttp] def encode: HttpHeaders =
-    self.toList
-      .foldLeft[HttpHeaders](new CombinedHttpHeaders(true)) { case (headers, entry) =>
+  private[zhttp] def encode: HttpHeaders = {
+    val (exceptions, regularHeaders) = self.toList.span(h => h._1.contains(HeaderNames.setCookie))
+    val combinedHeaders              = regularHeaders
+      .groupBy(_._1)
+      .map { case (key, tuples) =>
+        key -> tuples.map(_._2).map(value => if (value.contains(",")) s"""\"$value\"""" else value).mkString(",")
```

Comment:
Since you are already on it, please use default Netty methods, that supports `Iterable` with proper escape mechanism, thanks!
https://github.com/netty/netty/blob/3efd9642ca17b9470f38eea8d100e4cec9c41291/codec-http/src/main/java/io/netty/handler/codec/http/CombinedHttpHeaders.java#L168-L170

Example
https://scastie.scala-lang.org/8lW5DwGYSh6HcJnyXCgNUA

Reply:
Just went away from `CombinedHttpHeaders` as it has multiple issues.